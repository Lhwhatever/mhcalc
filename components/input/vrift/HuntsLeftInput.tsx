import { TextField } from '@material-ui/core'
import React from 'react'
import { InputEvent } from '../types'

export interface HuntsLeftInputProps {
    fullWidth?: boolean
    value?: number
    onChange?: (event: InputEvent) => void
}

const HuntsLeftInput = (props: HuntsLeftInputProps): JSX.Element => {
    return <TextField label="Hunts Left" type="number" variant="outlined" {...props} />
}

export default HuntsLeftInput
