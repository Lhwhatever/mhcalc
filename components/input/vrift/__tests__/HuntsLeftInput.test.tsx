import { TextField } from '@material-ui/core'
import { createShallow } from '@material-ui/core/test-utils'
import React from 'react'
import HuntsLeftInput from '../HuntsLeftInput'

describe('HuntsLeftInput test', () => {
    let shallow
    let wrapper

    beforeAll(() => {
        shallow = createShallow()
    })

    beforeEach(() => {
        wrapper = shallow(<HuntsLeftInput />)
    })

    it('should use TextField[label="Hunts Left"][type=number]', () => {
        expect(wrapper.find(TextField)).toHaveLength(1)
        expect(wrapper.find(TextField).prop('type')).toStrictEqual('number')
        expect(wrapper.find(TextField).prop('label')).toMatch(/hunts left/i)
    })

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
