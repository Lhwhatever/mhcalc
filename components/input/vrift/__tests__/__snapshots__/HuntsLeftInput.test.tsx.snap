// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HuntsLeftInput test should match snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  label="Hunts Left"
  type="number"
  variant="outlined"
/>
`;
